src/lib/models/abstraction-period.js

```diff
@@ -1,7 +1,55 @@
+'use strict';
+
+const { CHARGE_SEASON } = require('./constants');
 const Model = require('./model');
 const { assertDay, assertMonth } = require('./validators');
 
+const moment = require('moment');
```

Comment:
There could be shorter solution to this if we used moment range which we've used elsewhere, e.g.:
```
const moment = require('moment-range').extendMoment(require('moment'));

/**
 * Creates a moment range from the supplied abs period in the specified year
 * @param {AbstractionPeriod} abstractionPeriod
 * @param {Number} startYear
 * @return {MomentRange}
 */
const createRange = (abstractionPeriod, startYear) => {
  const { startDay, endDay, startMonth, endMonth } = abstractionPeriod;
  const isCrossYear = (startMonth > endMonth || (startMonth === endMonth && startDay > endDay));
  const endYear = isCrossYear ? startYear + 1 : startYear;
  return moment.range(
    `${startYear}-${startMonth}-${startDay}`,
    `${endYear}-${endMonth}-${endDay}`
  );
};

...

  /**
   * Checks if the passed AbstractionPeriod contains this instance. In order
   * for this to return true, this instance must fit inside the passed
   * AbstractionPeriod, but not matching the bounaries.
   *
   * @param {AbstractionPeriod} period The abstraction period to check if this instance fits within
   */
  isWithinAbstractionPeriod (period) {
    const thisRange = createRange(this, 2018);
    const testRanges = [createRange(period, 2017), createRange(period, 2018)];
    return testRanges.some(range => range.contains(thisRange));
  }
```